src/Model/Table/RequestsTable.php

```diff
@@ -79,22 +80,23 @@ public function gc()
         if (time() % 100 !== 0) {
             return;
         }
-        $purge = $this->find()
+        $noPurge = $this->find()
             ->select(['id'])
             ->hydrate(false)
-            ->where(['requested_at <=' => new \DateTime('-2 weeks')])
+            ->order(['requested_at' => 'desc'])
+            ->limit(Configure::read('DebugKit.requestCount') ?: 20)
```

Comment:
As it stands the history panel will only ever show at most 10 requests, so unless you would like to delve into the serialized mess that is the raw db I don't think it makes sense to keep more than 20 (At least by default).